sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -500,6 +499,19 @@ case class WholeStageCodegenRDD(@transient sc: SparkContext, var source: CodeAnd
 
   override def compute(split: Partition,
       context: TaskContext): Iterator[InternalRow] = {
+    try {
+      computeInternal(split, context)
+    } catch {
+      case ex: ClassCastException => {
+        // Any other exception in future might be added here
+        CodeGenerator.refresh(source)
+        computeInternal(split, context)
```

Comment:
Also, the cases this will solve only the case where a plan is trying to get an object from input adapter row.  e.g. com.blah.Event event = (com.blah.Event) inputRow.get(0).
Even if the Event class has some complex object structure and contains other objects which might be a potential candidate for CCE, that will happen before processing any row. Hence I feel it would be safe to assume not a single row has been processed.